pkg/policy/tree.go

```diff
@@ -89,45 +89,58 @@ func canConsume(root *Node, ctx *SearchContext) api.ConsumableDecision {
 
 // AllowsRLocked checks the ConsumableDecision of the given `SearchContext`.
 // Must be called with the Mutex's tree at least RLocked.
```

Comment:
It looks like it makes sense to have both, an `Allows` which locks and a `AllowsRLocked` for when he lock is already being held.